src/main/java/com/google/sps/servlets/DisplayRepliedPostsServlet.java

```diff
@@ -0,0 +1,51 @@
+package com.google.sps.servlets;
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.cloud.datastore.StructuredQuery.OrderBy;
+import com.google.cloud.datastore.Key;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for displaying posts. */
+@WebServlet("/display-posts")
+public class DisplayRepliedPostsServlet extends HttpServlet {
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        // Initialize Datastore 
+        Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+
+        Query<Entity> query = Query.newEntityQueryBuilder().setKind("RepliedPost").build();
+        QueryResults<Entity> results = datastore.run(query);
+
+        // Add the post content string into the 'posts' arraylist 
+        List<List<String>> posts = new ArrayList<>();
+        while (results.hasNext()) {
+            List<String> onePost = new ArrayList<>(); 
+            Entity entity = results.next();
+            String postContent = entity.getString("originalcontent");
+            String repliedContent = entity.getString("content");
```

Comment:
I don't fully understand the distinction between a RepliedPost and UnrepliedPost. Is the former a post with a bunch of comments/replies from other users, while the latter is a post that has no replies/comments.

If so, I am not sure if we should have different "kinds" of entities for them. We can try to do something like this: https://cloud.google.com/appengine/docs/standard/java/datastore/creating-entity-keys#Ancestor_paths, where we have a "Post" as the root entity and "Reply" as a child entity. We can discuss this further in our meeting.